cranelift/codegen/src/clif.isle

```diff
@@ -0,0 +1,1635 @@
+)
+
+(decl brif (IntCC Value Block ValueSlice) Inst)
+(extractor
+    (brif Cond f block args)
+    (inst_data (InstructionData.BranchInt (Opcode.Brif) Cond block (unwrap_head_value_list_1 f args)))
+)
+
+(decl brff (FloatCC Value Block ValueSlice) Inst)
+(extractor
+    (brff Cond f block args)
+    (inst_data (InstructionData.BranchFloat (Opcode.Brff) Cond block (unwrap_head_value_list_1 f args)))
+)
+
+(decl br_table (Value Block JumpTable) Inst)
+(extractor
+    (br_table x block JT)
+    (inst_data (InstructionData.BranchTable (Opcode.BrTable) block JT x))
+)
+
+(decl debugtrap () Inst)
+(extractor
+    (debugtrap )
+    (inst_data (InstructionData.NullAry (Opcode.Debugtrap)))
+)
+
+(decl trap (TrapCode) Inst)
+(extractor
+    (trap code)
+    (inst_data (InstructionData.Trap (Opcode.Trap) code))
+)
+
+(decl trapz (Value TrapCode) Inst)
+(extractor
+    (trapz c code)
+    (inst_data (InstructionData.CondTrap (Opcode.Trapz) code c))
+)
+
+(decl resumable_trap (TrapCode) Inst)
+(extractor
+    (resumable_trap code)
+    (inst_data (InstructionData.Trap (Opcode.ResumableTrap) code))
+)
+
+(decl trapnz (Value TrapCode) Inst)
+(extractor
+    (trapnz c code)
+    (inst_data (InstructionData.CondTrap (Opcode.Trapnz) code c))
+)
+
+(decl resumable_trapnz (Value TrapCode) Inst)
+(extractor
+    (resumable_trapnz c code)
+    (inst_data (InstructionData.CondTrap (Opcode.ResumableTrapnz) code c))
+)
+
+(decl trapif (IntCC Value TrapCode) Inst)
+(extractor
+    (trapif Cond f code)
+    (inst_data (InstructionData.IntCondTrap (Opcode.Trapif) Cond code f))
+)
+
+(decl trapff (FloatCC Value TrapCode) Inst)
+(extractor
+    (trapff Cond f code)
+    (inst_data (InstructionData.FloatCondTrap (Opcode.Trapff) Cond code f))
+)
+
+(decl return (ValueSlice) Inst)
+(extractor
+    (return rvals)
+    (inst_data (InstructionData.MultiAry (Opcode.Return)))
+)
+
+(decl fallthrough_return (ValueSlice) Inst)
+(extractor
+    (fallthrough_return rvals)
+    (inst_data (InstructionData.MultiAry (Opcode.FallthroughReturn)))
+)
+
+(decl call (FuncRef ValueSlice) Inst)
+(extractor
+    (call FN args)
+    (inst_data (InstructionData.Call (Opcode.Call) FN))
+)
+
+(decl call_indirect (SigRef Value ValueSlice) Inst)
+(extractor
+    (call_indirect SIG callee args)
+    (inst_data (InstructionData.CallIndirect (Opcode.CallIndirect) SIG (unwrap_head_value_list_1 callee args)))
+)
+
+(decl func_addr (FuncRef) Inst)
+(extractor
+    (func_addr FN)
+    (inst_data (InstructionData.FuncAddr (Opcode.FuncAddr) FN))
+)
+
+(decl splat (Value) Inst)
+(extractor
+    (splat x)
+    (inst_data (InstructionData.Unary (Opcode.Splat) x))
+)
+
+(decl swizzle (Value Value) Inst)
+(extractor
+    (swizzle x y)
+    (inst_data (InstructionData.Binary (Opcode.Swizzle) (value_array_2 x y)))
+)
+
+(decl insertlane (Value Value Uimm8) Inst)
+(extractor
+    (insertlane x y Idx)
+    (inst_data (InstructionData.TernaryImm8 (Opcode.Insertlane) Idx (value_array_2 x y)))
```

Comment:
I was poking around with a lowering that used this locally, and I think that this may not have been listed in the right order? I think that `Idx` should come after the `value_array_2`, right?

Reply:
I ended up having this -- https://github.com/alexcrichton/wasmtime/commit/d0f2bf4c140730566d00319def57fa1bdbc27269 -- to get matchers on `insertlane` working (although I'm not 100% sure it was correct)

Reply:
That looks correct to me. Do you want to make a PR after this one merges?